usaspending_api/disaster/helpers/covid_download_csv_strategies.py

```diff
@@ -0,0 +1,197 @@
+from abc import ABC, abstractmethod
+import multiprocessing
+import time
+import logging
+from pathlib import Path
+from typing import Tuple
+from django.conf import settings
+
+from usaspending_api.common.csv_helpers import count_rows_in_delimited_file
+from usaspending_api.common.helpers.s3_helpers import delete_s3_object, download_s3_object
+from usaspending_api.common.helpers.sql_helpers import read_sql_file_to_text
+from usaspending_api.download.filestreaming.download_generation import (
+    EXCEL_ROW_LIMIT,
+    split_and_zip_data_files,
+    wait_for_process,
+    execute_psql,
+    generate_export_query_temp_file,
+)
+from usaspending_api.download.filestreaming.zip_file import append_files_to_zip_file
+from usaspending_api.download.lookups import FILE_FORMATS
+from pyspark.sql import SparkSession
+from usaspending_api.common.etl.spark import hadoop_copy_merge, write_csv_file
+from usaspending_api.common.helpers.spark_helpers import configure_spark_session, get_active_spark_session
+from typing import List
+
+
+class AbstractToCSVStrategy(ABC):
```

Comment:
Since this just contains 1 function ... it's like a Single Abstract Method (SAM) interface, aka "functional interface" ... and may be able to be simplified in python by just defining the alternate functions and passing them in at the point of execution rather than having to instantiate the concrete strategy classes and calling their function. Food for thought.

Reply:
No it's a strategy. I'm using object oriented programming. Is it better to not do so?